### DISCUSSION TITLE
Does brakeman has any documentation that they follow Security Standards e.g. OWASP Top 10?

### DISCUSSION THREAD

**Discussion by fshaikh-sephora** [presidentbeef/brakeman#1802] [Q&A]
After looking around https://brakemanscanner.org/ documentation, could not find any references to practices brakeman following the security standards like OWASP Top 10. Could anyone share the link please?

Have search and found below so far:
- https://brakemanscanner.org/docs/warning_types/
- https://github.com/search?q=repo%3Apresidentbeef%2Fbrakeman%20OWASP&type=code

**Comment by presidentbeef** [comment]
[I wrote this a long time ago](https://stackoverflow.com/questions/35210138/ruby-on-rails-brakeman-gem-and-owasp-top-10), could use some updating.

**Reply by presidentbeef** [reply]
Oh also if you use the JSON report, it includes CWEs, which could be mapped to OWASP Top 10

```json
  "warnings": [
    {
      "warning_type": "SQL Injection",
      "warning_code": 0,
      "fingerprint": "02ad62a4e0cc17d972701be99e1d1ba4761b9176acc36e41498eac3a8d853a8a",
      "check_name": "SQL",
      "message": "Possible SQL injection",
      "file": "app/controllers/users_controller.rb",
      "line": 66,
      "link": "https://brakemanscanner.org/docs/warning_types/sql_injection/",
      "code": "@user.delete_by(params[:user])",
      "render_path": null,
      "location": {
        "type": "method",
        "class": "UsersController",
        "method": "destroy_them_all"
      },
      "user_input": "params[:user]",
      "confidence": "Medium",
      "cwe_id": [
        89
      ]
    },
...]
```
